python_client/src/wellcome_storage_service/downloader.py

```diff
@@ -57,15 +59,25 @@ def download_compressed_bag(storage_manifest, out_path):
     location = storage_manifest["location"]
     provider = _choose_provider(location)
 
+    ext_identifier = storage_manifest["info"]["externalIdentifier"]
+
     with tarfile.open(out_path, "w:gz") as tf:
         for manifest_file in _all_files(storage_manifest):
             fileobj = provider.get_fileobj(
                 location=location, manifest_file=manifest_file
             )
 
-            tarinfo = tarfile.TarInfo(name=manifest_file["name"])
+            tarinfo = tarfile.TarInfo(
+                name=os.path.join(ext_identifier, manifest_file["name"])
+            )
             tarinfo.size = manifest_file["size"]
 
+            # Set the modified time to the creation dateb of the storage manifest
+            creation_date = dt.datetime.strptime(
```

Comment:
This discards the time zone info from the timestamp which results in offset timestamps on the tar info. Which I only noticed because Archivematica's  Django settings specify a US time zone. In python 3 we could use %z to parse the time zone but not in py 2. Can we force the datetime object to utc here if it's safe to assume all timestamps from the storage service will be UTC?

Reply:
We could, but these timestamps are already somewhat arbitrary – the storage manifest timestamp is when it's marked as done in the storage service, which can vary by hours depending on how much other stuff is in flight.

It would be nice to improve this, but I'm not too fussed about it.